From f7a4338f1b5ef03dca83ce44075e9d6e5897e037 Mon Sep 17 00:00:00 2001
From: Kamil Aronowski <kamil.aronowski@yahoo.com>
Date: Mon, 8 May 2023 09:28:24 +0200
Subject: [PATCH 16/35] Skip testing msleep()

In preparation for renaming msleep() to usleep(), in some cases tests
were failing due to a mismatch between our declaration of the usleep()
function and what is being provided by unistd.h. This change simply
makes our function declared only when not in a unit test environment.

Signed-off-by: Kamil Aronowski <kamil.aronowski@yahoo.com>
---
 include/console.h | 2 ++
 lib/console.c     | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/console.h b/include/console.h
index 0c4a513..2a29c2d 100644
--- a/include/console.h
+++ b/include/console.h
@@ -122,7 +122,9 @@ extern EFI_STATUS EFIAPI vdprint_(const CHAR16 *fmt, const char *file, int line,
 extern EFI_STATUS print_crypto_errors(EFI_STATUS rc, char *file, const char *func, int line);
 #define crypterr(rc) print_crypto_errors((rc), __FILE__, __func__, __LINE__)
 
+#ifndef SHIM_UNIT_TEST
 extern VOID msleep(unsigned long msecs);
+#endif
 
 /* This is used in various things to determine if we should print to the
  * console */
diff --git a/lib/console.c b/lib/console.c
index 6bbb8a7..c7ca3bc 100644
--- a/lib/console.c
+++ b/lib/console.c
@@ -743,11 +743,13 @@ setup_verbosity(VOID)
 	setup_console(-1);
 }
 
+#ifndef SHIM_UNIT_TEST
 VOID
 msleep(unsigned long msecs)
 {
 	BS->Stall(msecs);
 }
+#endif
 
 /* This is used in various things to determine if we should print to the
  * console */
-- 
2.39.0

